api/client/samples/similar_regions/similar_region.py

```diff
@@ -40,15 +44,22 @@ def _regions_avail_for_selection(self, region_properties):
         self._logger.info("{} regions are available for comparison.".format(len(regions)))
         return list(regions)
 
-    def _format_results(self, sim_regions, region_level_id, dists, metric_dists):
+    def _format_results(self, sim_regions, requested_region_level, dists, metric_dists):
         for ranking, sim_region_region_id in enumerate(sim_regions):
-            sim_region_region = self.client.lookup("regions", sim_region_region_id)
-            sim_region_name = self.client.lookup("regions", sim_region_region_id)["name"]
-            if region_level_id is not None and sim_region_region["level"] != region_level_id:
-                self._logger.info("not level %s %s" % (region_level_id, sim_region_name))
+            region_info = self.client.lookup("regions", sim_region_region_id)
+            region_level = region_info["level"]
+            sim_region_name = region_info["name"]
+            
+            if requested_region_level is not None and region_level != requested_region_level:
+                self._logger.info("not level %s %s" % (requested_region_level, sim_region_name))
                 continue
-            sim_region_name = self.client.lookup("regions", sim_region_region_id)["name"]
-            parent = next(self.client.lookup_belongs("regions", sim_region_region_id), {"name": ""})
+                
+            # Choose parent one level up from this region
+            parent = {"name": ""}
+            for r in self.client.lookup_belongs("regions", sim_region_region_id):
```

Comment:
If we want to turn this into a general API feature it is probably better to do this in a different branch where the best way to do this can be better investigated. For example, note that non-empty grandparent is only used if parent is not available - not sure if this is what we want to do in general. So prefer to leave this as is in this similar_region-specific branch - we can later change to use new API feature once available

Reply:
yes